### ISSUE TITLE
[v2.0.17] Incorrect trim on Locale

### ISSUE THREAD

**Issue by gnh1201** [directus/api#770]
I had mentioned a similar problem in #740.
Since there was no major inconvenience, I decided to use it as it is,
but I found that the problem has spread to the locale after the version upgrade.

the 'en-US' is now displayed as 'n-U' in my User Directory.
https://i.imgur.com/GEm8GKc.png

**Comment by ceruulean** [comment]
Check the datatype of your database column? Is it large enough to hold the data?

**Comment by WellingGuzman** [comment]
Have you tried using the app and then the API and see if the problem happens on both end?

This is odd bug that we would like to solve.

- How the HTTP request body looks like? Does the app sends the full string (`en-US`) and the api responds with `n-U`?
- Are you on MySQL, MariaDB or any similar databases?
- What browser are you using?
- Any custom hooks/interface installed?

I am still not being able to reproduce this issue, any steps that helps us reproduce this will be appreciated.

**Comment by benhaynes** [comment]
@rijkvanzanten — could this be an App/Interface issue? Have you seen this before?

**Comment by shealavington** [comment]
I've noticed the same issue while at work — the timezone is empty and the locale is partial.

The issue started after manually modifying the `timezone` default in `directus_users` to be `Europe/London`. I'm not saying that's the cause, there were a few other variables I changed, however this is what stands out to me; hopefully that helps.

This is possibly an app-issue as looking at the network request it's returning `locale: "en-US"` for my user.
When the page is loaded the options for locale are available, but n-U is entered, however I can click on en-US and it does seem to work correctly.

I think we're on MariaDB
Chrome Browser
No custom interfaces

> On this note, I've just been reminded of another issue I need to create an issue for : Creating a user but not selecting a status is inserting it into the database as status of: "raf", I'll see if I can create steps to replicate later unless you wanted to give it a quick try haha.

**Comment by rijkvanzanten** [comment]
Haven't seen this before.

> ...it into the database as status of: "raf",...

This "Draft" saving as "raf" does indicate a similar problem however.

**Comment by benhaynes** [comment]
Yeah, the `length` being too short would truncate the _end_ not the beginning. I think this trim issue persists.
